homeassistant/components/lock/homematic.py

```diff
@@ -0,0 +1,59 @@
+"""
+Support for HomeMatic switches.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.homematic/
+"""
+import logging
+from homeassistant.components.lock import LockDevice
+from homeassistant.components.homematic import HMDevice, ATTR_DISCOVER_DEVICES
+from homeassistant.const import (STATE_UNKNOWN, STATE_LOCKED, STATE_UNLOCKED)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['homematic']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the HomeMatic switch platform."""
+    if discovery_info is None:
+        return
+
+    devices = []
+    for conf in discovery_info[ATTR_DISCOVER_DEVICES]:
+        new_device = HMLock(conf)
+        devices.append(new_device)
```

Comment:
`devices.append(HMLock(conf))`

Reply:
I did it the same way like in switch.homematic